File: P50.java
public class P50 {
    int primes[], len, num_len, num, longest_num_len, i, j, k;

    public P50() {
        longest_num_len=0;
        primes = new int[80000];
        len=2;
        primes[0]=2;
        primes[1]=3;
        for (i=5;i<1000000;i+=2) {
            int prime=1;
            j = 3;
            while (prime == 1 && j<=Math.sqrt(i)) {
                if (i%j == 0) {prime=0;}
                j += 2;
            }
            if (prime == 1) {primes[len]=i; len++;}
        }
        System.out.println("Length: "+ len);
    }

    public void get() {
        int sum;
        for (i=0;i<len-1;i++) {
            sum = primes[i];
            num_len=1;
            j=i+1;
            while (j<len & (sum+primes[j]) < 1000000) {
                sum += primes[j];
                //System.out.println("i: " + i + ":: j: " +j + ":: sum: " +sum);
                num_len ++;
                k=0;
                int prime = 1;
                while (prime == 1 && primes[k]<=Math.sqrt(sum)) {
                    if (sum%primes[k]==0) {prime=0;}
                    k++;
                }
                if (prime == 1 && num_len>longest_num_len) {
                    longest_num_len = num_len;
                    num = sum;
                }
                j++;
            }
        }
        System.out.println(longest_num_len + ": "+ num);
    }
    public static void main(String args[]) {
        P50 p = new P50();
        p.get();
    }
}

File: P50.c
#include <stdio.h>
#include <math.h>
#include <stdbool.h>

int main() {
	long a[80000], i, j, k, longest, len = 2, latest;
	a[0] = 2;
	a[1] = 3;

	//get all primes
	for (i=5;i<1000;i+=2) {
		j=3;
		bool prime = true;
		while (j<=sqrt(i) && prime) {
			if (i%j == 0){prime = false;}
			else {j+=2;}
		}
		if (prime) {a[len] = i; len++; latest = i;}
	}

	
	long long sum;
	long length = 1, alsoLongest = 0;
	longest = 0;
	//get the desired num
	
	for (i=0;i<len-1;i++) {
		sum = a[i];
		j=i;
		bool prime = true;
		while ((++j)<=len && prime && ((sum += a[j])<1000)) {
			sum += a[j];
			//
			//chk for prime
			length ++;
			k=-1;
			while (prime && a[++k]<=sqrt(sum)) {
				if (sum%a[k] == 0) {prime = false;}
				/*
				if (!prime) {
					prime = true;
					k=-1;
					sum -=a[i];
				}
				*/
			}
			if (prime && sum>longest && length>alsoLongest && sum < 1000) {longest = sum; alsoLongest = length;}
			/*
			for (k=j+1;k<len;k++) {
				if (sum == a[k]) {longest = sum;}
			}
			*/
		}

	}
	printf ("%ld, %d", latest, len);
	printf ("\n %ld %lld",alsoLongest, longest);
	return 0;
}
